src/inc-lib-init.php

```diff
@@ -1084,336 +1084,525 @@ function lovd_getInstallURL ($bFull = true)
-                // Found more positions? Return false.
+    
+    // Making sure wild types are not provided with nucleotides
+    // (e.g. c.123A=, which should be c.123=).
+    if ($aResponse['type'] == '=' && similar_text($sVariant, 'ACTG')) {
+        if ($bCheckHGVS) {
+            return false;
+        }
+        $aResponse['warnings']['WBASESONWILDTYPE'] = 'Unnecessary nucleotides provided for variant type "=".';
+    }
+
+    
+    // Making sure no redundant '?'s are given as positions.
+    if ($aVariant['latest_start'] . $aVariant['latest_start'] .
+         $aVariant['earliest_end'] . $aVariant['latest_end'] == '????') {
+        $sQuestionMarkWarning = 'Please rewrite the positions (?_?)_(?_?) to ?.';
+        
+    } elseif (($aVariant['latest_start'] . $aVariant['earliest_end'] == '??')) {
+        $sQuestionMarkWarning =
+            'Please rewrite the positions (' . $aVariant['earliest_start'] . '_?)_(?_' . $aVariant['latest_end'] .
+            ') to (' . $aVariant['earliest_start'] . '_' . $aVariant['latest_end'] . ').'; // (1_?)_(?_2) > (1_2)
+        
+    } elseif ($aVariant['earliest_start'] . $aVariant['latest_start'] == '??' && $aVariant['earliest_end']) {
+        $sQuestionMarkWarning = 'Please rewrite the positions (?_?)_' . $aVariant['earliest_end'] .
+            ' to ?_' . $aVariant['earliest_end'] . '.'; // (?_?)_2 > ?_2
+
+    } elseif ($aVariant['earliest_end'] . $aVariant['latest_end'] == '??') {
+        $sQuestionMarkWarning = 'Please rewrite the positions ' . $aVariant['earliest_start'] . '_(?_?) to '
+            . $aVariant['earliest_start'] . '_?.'; // 1_(?_?) > 1_?
+    
+    } elseif ($aVariant['earliest_start'] . $aVariant['earliest_end'] == '??' &&
+                !$aVariant['latest_start'] && !$aVariant['latest_end']) {
+        $sQuestionMarkWarning = 'Please rewrite the positions ?_? to ?.';
+    
+    } elseif ($aVariant['earliest_start'] . $aVariant['latest_start'] == '??' &&
+                isset($aResponse['messages']['IPOSITIONRANGE'])) {
+        $sQuestionMarkWarning = 'Please rewrite the positions (?_?) to ?.';
+    }
+    
+    if (isset($sQuestionMarkWarning)) {
+        if ($bCheckHGVS) {
+            return false;
+        }
+        $aResponse['warnings']['WTOOMUCHUNKNOWN'] = 'Redundant question marks were found. ' . $sQuestionMarkWarning;
+    }
+ 
+    
+    // Checking all type-specific format requirements.
+    if ($aVariant['type'] == 'delins' && !$aVariant['earliest_end'] && strlen($aVariant['suffix']) == 1) {
+        // If an insertion/deletion deletes one base and replaces it by one, it should be called and
+        //  formatted as a substitution.
+        if ($bCheckHGVS) {
+            return false;
+        }
+        $aResponse['warnings']['WWRONGTYPE'] = 'A deletion-insertion of one base to one base should be a substitution.';
+        
+    } elseif ($aVariant['type'] == 'ins') {
```

Comment:
Please add comments and examples to all code blocks below, I'm getting lost :sweat_smile: